File: src/main/java/org/retrolauncher/backend/app/games/domain/repositories/GameRepository.java
package org.retrolauncher.backend.app.games.domain.repositories;

import org.retrolauncher.backend.app._shared.domain.repository.Repository;
import org.retrolauncher.backend.app.games.domain.entities.Game;

import java.util.List;
import java.util.Optional;
import java.util.Set;
import java.util.UUID;

public interface GameRepository extends Repository<Game> {
    boolean existsByPath(String path);

    Optional<Game> findOneByNameAndPlatformId(String name, UUID platformId);

    List<Game> findAllByIdsNotIn(Set<String> exceptionsIds);

    void delete(Game game);
}

File: backend/src/app/games/domain/game_repository.go
package domain

import "retrolauncher/backend/src/app/games/domain/game"

type GameRepository interface {
	Save(entity *game.Game) error
	Get(id string) (*game.Game, error)
	List(input ListGamesParams) []game.Game
}

type ListGamesParams struct {
	Name string
}
